### PR TITLE
Add test for Delta XDS vs SotW

### PR DESCRIPTION
**Pull request by howardjohn**
This adds a new test -- disabled by default -- which adds a bunch of consistency checks in Delta XDS by comparing responses to SotW responses. This is intended to have 0 impact on production code, as all changes are behind a `UNSAFE` flag.

These tests have found a number of bugs in the past, as I have maintained this in my fork for a while. I think its useful to upstream though.

### CODE REVIEW

File: pilot/pkg/xds/eds.go
Comment (howardjohn) [inline pilot/pkg/xds/eds.go:556]: NOTE: I built this on top of https://github.com/istio/istio/pull/37974 to make things simpler. I will back it out if we don't merge that, and put a hold on this PR unti lthat one lands

File: pilot/pkg/model/context.go
Comment (hzxuzhonghu) [inline pilot/pkg/model/context.go:348]: LastResources

File: pilot/pkg/xds/delta.go
Comment (hzxuzhonghu) [inline pilot/pkg/xds/delta.go:719]: LGTM if move all these debug function to a separate file

**Review by adiprerepa — APPROVED** [review]
should we dump the delta results (bugs & potential optimizations) to a file of some sort? Could be an Istio report kind of thing, just thinking out loud here.

**Pull request merged**
